data/crac/crac-io/crac-io-json/src/main/java/com/powsybl/openrao/data/cracio/json/deserializers/PstRangeActionArrayDeserializer.java

```diff
@@ -156,4 +165,18 @@ private static void deserializeOlderOnConstraintUsageRules(JsonParser jsonParser
             throw new OpenRaoException("Unsupported field %s in CRAC version >= 2.4".formatted(keyword));
         }
     }
+
+    private static void deserializeVariationCosts(PstRangeActionAdder pstRangeActionAdder, JsonParser jsonParser) throws IOException {
+        while (jsonParser.nextToken() != JsonToken.END_OBJECT) {
+            if (UP.equals(jsonParser.getCurrentName())) {
+                jsonParser.nextToken();
+                pstRangeActionAdder.withVariationCost(jsonParser.getDoubleValue(), RangeAction.VariationDirection.UP);
+            } else if (DOWN.equals(jsonParser.getCurrentName())) {
+                jsonParser.nextToken();
+                pstRangeActionAdder.withVariationCost(jsonParser.getDoubleValue(), RangeAction.VariationDirection.DOWN);
```

Comment:
instead of using if/else, create a "deserialize" method (in the Constants class) that consumes a String and returns a VariationDirection